packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -1043,3 +1044,35 @@ class ExcludeFocus extends StatelessWidget {
     );
   }
 }
+
+/// A focus area is used to expand a focus node as part of the render object layer
+/// outside of its descendant render objects.
+///
+/// This is used by the [TextField] widgets to associate the focus node with the input
+/// decorations.
+class FocusArea extends SingleChildRenderObjectWidget {
+  /// Create a new [FocusArea] widget.
+  const FocusArea({required this.focusNode, required Widget child, Key? key}) : super(child: child, key: key);
+
+  /// The focus node this area encompasses.
+  final FocusNode focusNode;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return RenderFocusArea(focusNode);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant RenderFocusArea renderObject) {
+    renderObject.focusNode = focusNode;
+  }
+}
+
+/// The focus area render object is used as a sentinel in routes.dart.
```

Comment:
Maybe explain in more detail? and it feel weird to me that we point to a source code file in API doc.